temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -211,16 +210,37 @@ public void handle(PollWorkflowTaskQueueResponse task) throws Exception {
           workerMetricsScope.tagged(
               ImmutableMap.of(MetricsTag.WORKFLOW_TYPE, task.getWorkflowType().getName()));
 
+      String runId = task.getWorkflowExecution().getRunId();
+
       MDC.put(LoggerTag.WORKFLOW_ID, task.getWorkflowExecution().getWorkflowId());
       MDC.put(LoggerTag.WORKFLOW_TYPE, task.getWorkflowType().getName());
-      MDC.put(LoggerTag.RUN_ID, task.getWorkflowExecution().getRunId());
+      MDC.put(LoggerTag.RUN_ID, runId);
 
-      Lock runLock = null;
+      boolean locked = false;
       if (!Strings.isNullOrEmpty(stickyTaskQueueName)) {
-        runLock = runLocks.getLockForLocking(task.getWorkflowExecution().getRunId());
+        // Serialize workflow task processing for a particular workflow run.
+        // This is used to make sure that query tasks and real workflow tasks
+        // are serialized when sticky is on.
+        //
         // Acquiring a lock with a timeout to avoid having lots of workflow tasks for the same run
         // id waiting for a lock and consuming threads in case if lock is unavailable.
-        if (!runLock.tryLock(1, TimeUnit.SECONDS)) {
+        //
+        // Throws interrupted exception which is propagated. It's a correct way to handle it here.
+        //
+        // TODO 1: "1 second" timeout looks potentially too strict, especially if we are talking
+        //   about workflow tasks with local activities.
+        //   This could lead to unexpected fail of queries and reset of sticky queue.
+        //   Should it be connected to a workflow task timeout / query timeout?
+        // TODO 2: Does "consider increasing workflow task timeout" advice in this exception makes
+        //   any sense?
+        //   This MAYBE makes sense only if a previous workflow task timed out, it's still in
+        //   progress on the worker and the next workflow task got picked up by the same exact
+        //   worker from the general non-sticky task queue.
+        //   Even in this case, this advice looks misleading, something else is going on
+        //   (like an extreme network latency).
+        locked = runLocks.tryLock(runId, 1, TimeUnit.SECONDS);
+
+        if (!locked) {
```

Comment:
The change behind this line in this class is refactoring which makes the logic easier to follow, and may be ignored during the review.

Reply:
I think it should be changed to the workflow task timeout.

Reply:
Thanks for confirming that this 1 sec looks off! Let's discuss more and change it. While important to do, unrelated to the scope of this PR, we can do it separately.

Reply:
It's more to actually let the workflow task in flight to finish. Workflow task timeout is about 10s, deadlock detector is 1 second by default, seconds typically, there is more room to wait more than 1 second typically.